### PR TITLE
Replace `collectOnce` with `first`

### PR DESCRIPTION
**Pull request by KevinBoulongne**

### CODE REVIEW

**Review by JorisBodin — APPROVED** [review]

File: app/src/main/java/com/infomaniak/mail/data/MailData.kt [hunk 1]
```diff
@@ -112,29 +113,29 @@ object MailData {
 
     private fun loadAddressBooks(completion: () -> Unit) {
         CoroutineScope(Dispatchers.IO).launch {
-            MailRealm.readAddressBooks().collectOnce { realmAddressBooks ->
-                mutableAddressBooksFlow.value = realmAddressBooks
+            val realmAddressBooks = MailRealm.readAddressBooks().first().list.toList()
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/mail/data/MailData.kt:116]: I don't think that the `toList` is really needed

**Review by sirambd — CHANGES_REQUESTED** [review]
- [x] To simplify the code, we can use the advantages that realm offers us thanks to the `flow`, by making the database live thanks to a query as long as it's possible, and just update the database with the api, so that it automatically updates the ui.

File: app/src/main/java/com/infomaniak/mail/data/MailData.kt [hunk 2]
```diff
@@ -112,29 +113,29 @@ object MailData {
...
 
-                val apiAddressBooks = MailApi.fetchAddressBooks()
-                val mergedAddressBooks = mergeAddressBooks(realmAddressBooks, apiAddressBooks)
+            mutableAddressBooksFlow.value = realmAddressBooks
 
-                mutableAddressBooksFlow.value = mergedAddressBooks
+            val apiAddressBooks = MailApi.fetchAddressBooks()
+            val mergedAddressBooks = mergeAddressBooks(realmAddressBooks, apiAddressBooks)
```
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/data/MailData.kt:121]: Will be done in another PR, this refactor should be global for the whole app and not just for here.

File: app/src/main/java/com/infomaniak/mail/data/MailData.kt [hunk 3]
```diff
@@ -247,7 +249,8 @@ object MailData {
     }
 
     private fun getFoldersFromRealm() {
-        MailRealm.readFolders().collectOnce { realmFolders ->
+        CoroutineScope(Dispatchers.IO).launch {
+            val realmFolders = MailRealm.readFolders().first().list.toList()
 
             mutableFoldersFlow.value = realmFolders
```
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/data/MailData.kt]: Will be done in another PR, this refactor should be global for the whole app and not just for here.

**Pull request merged**